src/cpp/cpp_typecheck_compound_type.cpp

```diff
@@ -606,7 +606,7 @@ void cpp_typecheckt::typecheck_compound_declarator(
           id2string(component.get_name())+"::"+id2string(virtual_base);
         func_symb.base_name = component.get_base_name();
         func_symb.pretty_name = component.get_base_name();
-        func_symb.mode=ID_cpp;
+        func_symb.mode = symbol.mode;
```

Comment:
This one sounds worth testing to make sure the mode == C case is doing what you expect

Reply:
Let me actually remove this particular commit from the PR, it's the one that does deserve separate testing.